src/EasyAbp.BookingService.Domain/EasyAbp/BookingService/PeriodSchemes/PeriodScheme.cs

```diff
@@ -28,7 +29,41 @@ public PeriodScheme(Guid id, Guid? tenantId, string name, bool isDefault, List<P
         TenantId = tenantId;
         Name = name;
         IsDefault = isDefault;
-        
+
         Periods = periods ?? new List<Period>();
     }
+
+    public void UpdateIsDefault(bool v)
+    {
+        IsDefault = v;
+    }
+
+    public void Update(string name, List<Period> periods)
+    {
+        Name = name;
+        Periods = periods;
+    }
+
+    public TimeSpan GetLatestEndingTime(PeriodScheme effectivePeriodScheme)
+    {
+        TimeSpan endingTime;
+        if (effectivePeriodScheme.Periods.IsNullOrEmpty())
+        {
+            endingTime = TimeSpan.FromDays(1);
```

Comment:
I think when the Periods collection is empty, it should throw since the method's name is `GetLatestEndingTime`. Or you rename it to `FindLatestEndingTime` and get a `null` value.

How to understand the endingTime value is `24:00:00` by default?

Reply:
> How to understand the endingTime value is 24:00:00 by default?

`PeriodScheme` describes how `Periods` can be booked in **one day**, So, when there is no period, the end of this `PeriodScheme` is `24:00:00`

Reply:
`GetLatestEndingTime` method actually calculates the ending time of current `PeriodScheme`, to find applied `AssetSchedule` in next step.

Reply:
```csharp
    public TimeSpan? FindLatestEndingTime()
    {
        return Periods
            .Select(x => x.GetEndingTime())
            .OrderByDescending(x => x)
            .FirstOrDefault();
    }
```

```csharp
        var endingTime = effectivePeriodScheme.FindLatestEndingTime();

        // Fill the endingTime to 1 day.
        if (endingTime is null || endingTime.Value.Ticks < TimeSpan.TicksPerDay)
        {
            endingTime = TimeSpan.FromDays(1);
        }

        var endingDateTime = searchDate.Add(endingTime.Value);
```

What do you think?